### DISCUSSION TITLE
Goatcounter stopped working

### DISCUSSION THREAD

**Discussion by unbreakabl3** [cotes2020/jekyll-theme-chirpy#1569] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

Hello, dear all.
The integration with Goatcounter went smoothly, but since a few days ago, I cannot see new visitors anymore. I tried to find in the compiled `_site` any mention of goat-counter, but nothing. How can I check what is wrong or what the proper settings should be?

Thanks a lot in advance.

### Operations you have already tried

1. Re-deploy the site
2. Use VPN to simulate unique site visit
3. Clone vanilla Chirpy Starter template

**Comment by kungfux** [comment]
No much to share but here are some insights.

- I experience same issue starting from Feb 27
- GoatCounter continues to return page views as expected
- I'm able to count page visit through API

```sh
token=
api=https://kungfux.goatcounter.com/api/v0

curl -X POST "$api/count" \
    -H 'Content-Type: application/json' \
    -H "Authorization: Bearer $token" \
    --data '{"no_sessions": true, "hits": [{"path": "/about"}]}'
```
[API Docs](https://goatcounter.com/help/backend)
[Manage API tokens](https://kungfux.goatcounter.com/user/api)

**Reply by unbreakabl3** [reply]
Thanks a lot. I ran it and saw a spike in the graphs. 
When I open a new browser on another computer and go to the page, I can not see this access as a recent visit to Goatcounter (anymore). Do you have any idea why?

**Reply by kungfux** [reply]
I have no additional information as of now.

However, it seems like Google Analytics does not work too.

**Reply by cotes2020** [reply]
It seems that I made an oversight when packaging [`6.5.1`](https://rubygems.org/gems/jekyll-theme-chirpy/versions/6.5.1): I pre-released the code for web analytics configuration.

I'll be releasing a new version `6.5.2` to fix this later.

**Reply by kungfux** [reply]
Thanks for checking, Cotes.

I had an idea to check this but found no related changes in [changelog](https://github.com/cotes2020/jekyll-theme-chirpy/compare/v6.5.0...v6.5.1) and so did not try to play with configuration. I assume release process is a bit more complicated.

**Reply by cotes2020** [reply]
Packing the gem is done automatically by the script (`tools/release`), and this issue was a code leak caused by the Git branches I changed manually.

**Reply by unbreakabl3** [reply]
Thanks a lot for your fast support. So, I will wait for 6.5.2 to be released and check it again.

**Comment by kungfux** [comment, marked as answer]
@unbreakabl3 Meanwhile, analytics can be fixed by adding a new section to configuration:

```yaml
# Web Analytics Settings
analytics:
  google:
    id: # fill in your Google Analytics ID
  goatcounter:
    id: # fill in your GoatCounter ID
```

I would also recommend to leave existing analytics configuration untouched until major release.

You may receive errors while trying to publish the site and it will require to override `_includes/analytics/goatcounter.html` by creating this file:
```html
<!-- GoatCounter -->
<script
  async
  src="https://gc.zgo.at/count.js"
  data-goatcounter="https://{{ site.analytics.goatcounter.id }}.goatcounter.com/count"
></script>
```

**Reply by unbreakabl3** [reply]
Thank you. I will do it now and will update you.
PS. Version 6.5.2 doesn't fix that problem yet, right?

**Reply by unbreakabl3** [reply]
Confirm the provided solution fixes the problem. Thank you very much!

**Reply by kungfux** [reply]
> PS. Version 6.5.2 doesn't fix that problem yet, right?

Hopefully, it does. I had no chance to check yet as it was just released :)

**Reply by unbreakabl3** [reply]
So... For me, the 6.5.2 with the provided workaround is working. I just can't tell if it's working because of 6.5.2 or because of the workaround 😁.

**Reply by kungfux** [reply]
It works without any workarounds now.

**Comment by ablzh** [comment]
Apparently, the problem persists because simply specifying the ID in `_config.yml` wasn't enough.
So, I created a `goatcounter.html` file with the script specified in the goatcounter dashboard, pre-created the directories you specified, and now everything works.

Am I correct in understanding that the ID is the first part of the goatcounter URL (the Account name during registration?) I just couldn't find any ID on the GoatCounter website.

**Reply by kungfux** [reply]
Yes, GoatCounter's ID is called `code` .

`goatcounter.html` is already a [part](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/_includes/pageviews/goatcounter.html) of the Chirpy.

**Reply by ablzh** [reply]
Hi! Thank you!
I followed the instructions on the website, selecting [Option 2. Forking the Theme](https://chirpy.cotes.page/posts/getting-started/#option-2-forking-the-theme). I must have made a mistake somewhere because I didn't have the _include folder at all, and I'm probably missing something else.
